include/secp256k1.h

```diff
@@ -5,6 +5,83 @@
 extern "C" {
 # endif
 
+/** Opaque data structure that holds context information (precomputed tables etc.).
+ *
+ *  The purpose of context structures is to cache large precomputed data tables
+ *  that are expensive to construct, and also to maintain the randomization data
+ *  for blinding.
+ *
+ *  Do not create a new context object for each operation, as construction is
+ *  far slower than all other API calls (~100 times slower than an ECDSA
+ *  verification), and then use the context for all other operations.
+ *
+ *  A constructed context can safely be used from multiple threads
+ *  simultaneously, but some API calls need exclusive access, in particular
+ *  secp256k1_context_destroy and secp256k1_context_randomize.
```

Comment:
Do we want to drop the comment that `const` makes the difference between requiring exclusive access and not?